astropy/table/table.py

```diff
@@ -3674,3 +3683,42 @@ def __setstate__(self, state):
         nd_state, own_state = state
         super().__setstate__(nd_state)
         self.__dict__.update(own_state)
+
+
+class TableAttribute:
+    """
+    Descriptor to define custom Table attribute which gets stored in the object
+    meta dict and can have a defined default.
+
+    :param default: default value
+
+    """
+    def __init__(self, default=None):
+        self.default = default
+
+    def __get__(self, instance, owner):
+        try:
+            return instance.meta[self.name]
+        except AttributeError:
```

Comment:
This would also go wrong if the `TableAttribute` was attached to a class without a `meta`, leading to a difficult-to-debug state. I'd just do 
```
if instance is None:
    return self
```
which is very fast (and keeps the more relevant `KeyError` very close). Or, if you prefer the error route,
```
if instance is None:
    return self
else:
    raise
```